src/ln/peer_handler.rs

```diff
@@ -614,6 +611,7 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 													if self.initial_syncs_sent.load(Ordering::Acquire) < INITIAL_SYNCS_TO_SEND {
 														self.initial_syncs_sent.fetch_add(1, Ordering::AcqRel);
 														local_features.set_initial_routing_sync();
+														local_features.set_upfront_shutdown_script();
```

Comment:
Huh? We should set this on all peers, not just the first 5.

Reply:
Also, technically this hunk should go in a later commit, when we set ourselves to always send (at least) a 0-length script.

Reply:
Maybe add a test that connects a bunch of peers, too, while you're at it?

Reply:
Oh yes confused myself on this one! Move to right commit